src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -235,7 +235,29 @@ public async Task<ServerNode> GetCurrentSessionNode()
             SessionInfo = new SessionInfo(_clientSessionId, false, _documentStore.GetLastTransactionIndex(DatabaseName), options.NoCaching);
             TransactionMode = options.TransactionMode;
 
-            _javascriptCompilationOptions = new JavascriptCompilationOptions
+            var extensions = new JavascriptConversionExtension[]
+            {
+                JavascriptConversionExtensions.MathSupport.Instance,
+                new JavascriptConversionExtensions.DictionarySupport(),
+                JavascriptConversionExtensions.LinqMethodsSupport.Instance,
+                new JavascriptConversionExtensions.ConstSupport(Conventions),
+                JavascriptConversionExtensions.ToStringSupport.Instance,
+                JavascriptConversionExtensions.DateTimeSupport.Instance,
+                JavascriptConversionExtensions.InvokeSupport.Instance,
+                JavascriptConversionExtensions.NullCoalescingSupport.Instance,
+                JavascriptConversionExtensions.StringSupport.Instance,
+                JavascriptConversionExtensions.NestedConditionalSupport.Instance,
+                JavascriptConversionExtensions.ReservedWordsSupport.Instance,
+                JavascriptConversionExtensions.ValueTypeParseSupport.Instance,
+                JavascriptConversionExtensions.JsonPropertyAttributeSupport.Instance,
+                JavascriptConversionExtensions.NullComparisonSupport.Instance,
+                JavascriptConversionExtensions.NullableSupport.Instance,
+                JavascriptConversionExtensions.NewSupport.Instance,
+                JavascriptConversionExtensions.ListInitSupport.Instance,
+                MemberInitAsJson.ForAllTypes
+            };
+
```

Comment:
do we need all these extensions ?

Reply:
I see that in v5.0 we changed it to this:
https://github.com/ravendb/ravendb/blob/0c45f188594f7328a139184047a3738a53ba77a0/src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs#L232-L237

Reply:
Let's check if this passes in 5.0. If yes. Add a test there and close down the issue. Do not want to make that change in 4.2

Reply:
done